code/game/objects/items/devices/PDA/utilities.dm

```diff
@@ -0,0 +1,192 @@
+
+/datum/data/pda/utility/flashlight/start()
+	fon = !fon
+	name = fon ? "Disable Flashlight" : "Enable Flashlight"
+	pda.update_shortcuts()
+	pda.set_light(fon ? f_lum : 0)
+
+/datum/data/pda/utility/honk
+	name = "Honk Synthesizer"
+	icon = "smile-o"
+	category = "Clown"
+
+	var/last_honk //Also no honk spamming that's bad too
+
+/datum/data/pda/utility/honk/start()
+	if(!(last_honk && world.time < last_honk + 20))
+		playsound(pda.loc, 'sound/items/bikehorn.ogg', 50, 1)
+		last_honk = world.time
+
+/datum/data/pda/utility/toggle_door
+	name = "Toggle Door"
+	icon = "external-link"
+	var/remote_door_id = ""
+
+/datum/data/pda/utility/toggle_door/start()
+	for(var/obj/machinery/door/poddoor/M in world)
+		if(M.id_tag == remote_door_id)
+			if(M.density)
+				M.open()
+			else
+				M.close()
+
+/datum/data/pda/utility/scanmode/medical
+	base_name = "Med Scanner"
+	icon = "heart-o"
+
+/datum/data/pda/utility/scanmode/medical/scan_mob(mob/living/C as mob, mob/living/user as mob)
+	for (var/mob/O in viewers(C, null))
+		O.show_message("<span class=warning>[user] has analyzed [C]'s vitals!</span>", 1)
+
+	user.show_message("<span class=notice>Analyzing Results for [C]:</span>")
+	user.show_message("<span class=notice>\t Overall Status: [C.stat > 1 ? "dead" : "[C.health - C.halloss]% healthy"]</span>", 1)
+	user.show_message("<span class=notice>\t Damage Specifics: [C.getOxyLoss() > 50 ? "</span><span class=warning>" : ""][C.getOxyLoss()]-[C.getToxLoss() > 50 ? "</span><span class=warning>" : "</span><span class=notice>"][C.getToxLoss()]-[C.getFireLoss() > 50 ? "</span><span class=warning>" : "</span><span class=notice>"][C.getFireLoss()]-[C.getBruteLoss() > 50 ? "</span><span class=warning>" : "</span><span class=notice>"][C.getBruteLoss()]</span>", 1)
+	user.show_message("<span class=notice>\t Key: Suffocation/Toxin/Burns/Brute</span>", 1)
+	user.show_message("<span class=notice>\t Body Temperature: [C.bodytemperature-T0C]&deg;C ([C.bodytemperature*1.8-459.67]&deg;F)</span>", 1)
+	if(C.timeofdeath && (C.stat == DEAD || (C.status_flags & FAKEDEATH)))
+		user.show_message("<span class=notice>\t Time of Death: [C.timeofdeath]</span>")
+	if(istype(C, /mob/living/carbon/human))
+		var/mob/living/carbon/human/H = C
+		var/list/damaged = H.get_damaged_organs(1,1)
+		user.show_message("<span class=notice>Localized Damage, Brute/Burn:</span>",1)
+		if(length(damaged)>0)
+			for(var/obj/item/organ/external/org in damaged)
+				user.show_message("<span class=notice>\t [capitalize(org.name)]: [org.brute_dam > 0 ? "<span class=warning>[org.brute_dam]</span>" : "0"]-[org.burn_dam > 0 ? "<span class=warning>[org.burn_dam]</span>" : "0"]", 1)
+		else
+			user.show_message("<span class=notice>\t Limbs are OK.",1)
+
+/datum/data/pda/utility/scanmode/dna
+	base_name = "DNA Scanner"
+	icon = "link"
+
+/datum/data/pda/utility/scanmode/dna/scan_mob(mob/living/C as mob, mob/living/user as mob)
+	if (!istype(C:dna, /datum/dna))
```

Comment:
This is AFTER I made a lot of the scanner code less gross. (It was really gross.)